data.go

```diff
@@ -0,0 +1,61 @@
+package gotorch
+
+// #cgo CFLAGS: -I ${SRCDIR}/cgotorch
+// #cgo LDFLAGS: -L ${SRCDIR}/cgotorch -Wl,-rpath ${SRCDIR}/cgotorch -lcgotorch
+// #cgo LDFLAGS: -L ${SRCDIR}/cgotorch/libtorch/lib -Wl,-rpath ${SRCDIR}/cgotorch/libtorch/lib -lc10 -ltorch -ltorch_cpu
+// #include <stdlib.h>
+// #include "cgotorch.h"
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Dataset interface
+type Dataset struct {
+	T C.CDataset
+}
+
+// Transform interface
+type Transform interface{}
```

Comment:
No method in this interface type? Then, why do we need it to be an interface?

Reply:
The interface is used in `AddTransofrm` function, which accepts various transform instance:
``` golang
func (d *Dataset) AddTransforms(transforms []Transform) {
...
}

dataset.AddTransforms([]torch.Transform{
		torch.NewNormalize(0.1307, 0.3081),
		torch.NewStack(),
	})

```

Reply:
Could Go programmers define transformers in Go? If so, what methods they are supposed to implement?